src/Infrastructure.Dapper/AdminConsole/Helpers/BulkResourceCreationService.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Data;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Microsoft.Data.SqlClient;
+
+namespace Bit.Infrastructure.Dapper.AdminConsole.Helpers;
+
+public static class BulkResourceCreationService
+{
+    private const string _defaultErrorMessage = "Must have at least one record for bulk creation.";
+    public static async Task CreateCollectionsUsersAsync(SqlConnection connection, SqlTransaction transaction, IEnumerable<CollectionUser> collectionUsers, string errorMessage = _defaultErrorMessage)
+    {
+        using var bulkCopy = new SqlBulkCopy(connection, SqlBulkCopyOptions.KeepIdentity, transaction);
+        bulkCopy.DestinationTableName = "[dbo].[CollectionUser]";
+        var dataTable = BuildCollectionsUsersTable(bulkCopy, collectionUsers, errorMessage);
+        await bulkCopy.WriteToServerAsync(dataTable);
+    }
+
+    private static DataTable BuildCollectionsUsersTable(SqlBulkCopy bulkCopy, IEnumerable<CollectionUser> collectionUsers, string errorMessage)
+    {
+        var collectionUser = collectionUsers.First();
```

Comment:
I suggest `FirstOrDefault` so that an empty sequence can be handled by your null check below, rather than having `First` throw an error.